src/minisketch/include/minisketch.h

```diff
@@ -0,0 +1,367 @@
+     *  If a particular field size `bits` is supported, implementation 0 is always supported for it.
+     *  Higher implementation numbers may or may not be available as well, up to MaxImplementation().
+     */
+    static bool ImplementationSupported(uint32_t bits, uint32_t implementation) noexcept { return minisketch_implementation_supported(bits, implementation); }
+
+    /** Given field size and a maximum number of decodable elements n, compute what capacity c to
+     *  use so that sketches with more elements than n have a chance no higher than 2^-fpbits of
+     *  being decoded incorrectly (and will instead fail when decoding for up to n elements).
+     *
+     *  See minisketch_compute_capacity for more details. */
+    static size_t ComputeCapacity(uint32_t bits, size_t max_elements, uint32_t fpbits) noexcept { return minisketch_compute_capacity(bits, max_elements, fpbits); }
+
+    /** Reverse operation of ComputeCapacity. See minisketch_compute_max_elements. */
+    static size_t ComputeMaxElements(uint32_t bits, size_t capacity, uint32_t fpbits) noexcept { return minisketch_compute_max_elements(bits, capacity, fpbits); }
+
+    /** Construct a clone of the specified sketch. */
+    Minisketch(const Minisketch& sketch) noexcept
+    {
+        if (sketch.m_minisketch) {
+            m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_clone(sketch.m_minisketch.get()));
+        }
+    }
+
+    /** Make this Minisketch a clone of the specified one. */
+    Minisketch& operator=(const Minisketch& sketch) noexcept
+    {
+        if (sketch.m_minisketch) {
+            m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_clone(sketch.m_minisketch.get()));
+        }
+        return *this;
+    }
+
+    /** Check whether this Minisketch object is valid. */
+    explicit operator bool() const noexcept { return bool{m_minisketch}; }
+
+    /** Construct an (invalid) Minisketch object. */
+    Minisketch() noexcept = default;
+
+    /** Move constructor. */
+    Minisketch(Minisketch&&) noexcept = default;
+
+    /** Move assignment. */
+    Minisketch& operator=(Minisketch&&) noexcept = default;
+
+    /** Construct a Minisketch object with the specified parameters.
+     *
+     * If bits is not BitsSupported(), or the combination of bits and capacity is not
+     * ImplementationSupported(), or OOM occurs internally, an invalid Minisketch
+     * object will be constructed. Use operator bool() to check that this isn't the
+     * case before performing any other operations. */
+    Minisketch(uint32_t bits, uint32_t implementation, size_t capacity) noexcept
+    {
+        m_minisketch = std::unique_ptr<minisketch, Deleter>(minisketch_create(bits, implementation, capacity));
+    }
+
+    /** Create a Minisketch object sufficiently large for the specified number of elements at given fpbits.
+     *  It may construct an invalid object, which you may need to check for. */
```

Comment:
nit: make the "may" a "must", at least I interpret the requirement on the non-wrapped construction already as a must "Use operator bool() to check that this isn't the case before performing any other operations"

Reply:
See the comment above, which describes the conditions under which an invalid object may be constructed. There are reasonable use cases where you don't need to check (you already made sure the fieldsize/implementation combination is supported, and you're ok with treating OOM as a fatal error).